<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<title>换脸插件升级导致SDWebUI无法启动cannot import name ‘Undefined‘ from ‘pydantic.fields‘ - 编程大咖</title>
	<script>(function(d,e){d[e]=d[e].replace("no-js","js");})(document.documentElement,"className");</script>
	<meta name="description" content="">
		<meta property="og:url" content="https://bcdaka.github.io/posts/1e6542fffe887c8ff50d91edb13e448b/">
  <meta property="og:site_name" content="编程大咖">
  <meta property="og:title" content="换脸插件升级导致SDWebUI无法启动cannot import name ‘Undefined‘ from ‘pydantic.fields‘">
  <meta property="og:description" content="今天在一台新的机器环境装了SDWEBUI，都使用最新的版本，升级了下换脸的插件，于是乎启动崩溃了。错误如下
Launching Web UI with arguments: --listen --skip-torch-cuda-test --disable-nan-check --skip-version-check --skip-python-version-check --enable-insecure-extension-access --api --xformers --xformers-flash-attention --opt-sdp-attention --no-half-vae --ad-no-huggingface --theme=dark --lora-dir /mnt/e/sddata/Lora --vae-dir /mnt/e/sddata/VAE --ckpt-dir /mnt/e/sddata/checkpoint --controlnet-dir /mnt/e/sddata/ControlNet --embeddings-dir /mnt/e/sddata/embeddings --controlnet-annotator-models-path /mnt/e/sddata/downloads Traceback (most recent call last): File &#34;/mnt/e/sd-webui-aki-v4.7/launch.py&#34;, line 48, in &lt;module&gt; main() File &#34;/mnt/e/sd-webui-aki-v4.7/launch.py&#34;, line 44, in main start() File &#34;/mnt/e/sd-webui-aki-v4.7/modules/launch_utils.py&#34;, line 465, in start import webui File &#34;/mnt/e/sd-webui-aki-v4.7/webui.py&#34;, line 13, in &lt;module&gt; initialize.imports() File &#34;/mnt/e/sd-webui-aki-v4.7/modules/initialize.py&#34;, line 23, in imports import gradio # noqa: F401 ^^^^^^^^^^^^^ File &#34;">
  <meta property="og:locale" content="zh_CN">
  <meta property="og:type" content="article">
    <meta property="article:section" content="posts">
    <meta property="article:published_time" content="2024-04-23T15:39:00+08:00">
    <meta property="article:modified_time" content="2024-04-23T15:39:00+08:00">

	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
	<link rel="dns-prefetch" href="//fonts.googleapis.com">
	<link rel="dns-prefetch" href="//fonts.gstatic.com">
	<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700">

	<link rel="stylesheet" href="/css/style.css">
	

	<link rel="shortcut icon" href="/favicon.ico">
		
  


</head>
<body class="body">
	<div class="container container--outer">
		<header class="header">
	<div class="container header__container">
		
	<div class="logo">
		<a class="logo__link" href="/" title="编程大咖" rel="home">
			<div class="logo__item logo__text">
					<div class="logo__title">编程大咖</div>
					
				</div>
		</a>
	</div>
		<div class="divider"></div>
	</div>
</header>
		<div class="wrapper flex">
			<div class="primary">
			
<main class="main" role="main">
	<article class="post">
		<header class="post__header">
			<h1 class="post__title">换脸插件升级导致SDWebUI无法启动cannot import name ‘Undefined‘ from ‘pydantic.fields‘</h1>
			
		</header>
		<div id="gatop"></div>
		<div class="content post__content clearfix">
			
<div id="content_views" class="htmledit_views">
                    <p style="text-align:center;"><img alt="" src="https://images2.imgbox.com/83/15/Is9wcSOF_o.png"></p> 
<p>今天在一台新的机器环境装了SDWEBUI，都使用最新的版本，升级了下换脸的插件，于是乎启动崩溃了。错误如下</p> 
<pre><code class="language-bash">Launching Web UI with arguments: --listen --skip-torch-cuda-test --disable-nan-check --skip-version-check --skip-python-version-check --enable-insecure-extension-access --api --xformers --xformers-flash-attention --opt-sdp-attention --no-half-vae --ad-no-huggingface --theme=dark --lora-dir /mnt/e/sddata/Lora --vae-dir /mnt/e/sddata/VAE --ckpt-dir /mnt/e/sddata/checkpoint --controlnet-dir /mnt/e/sddata/ControlNet --embeddings-dir /mnt/e/sddata/embeddings --controlnet-annotator-models-path /mnt/e/sddata/downloads
Traceback (most recent call last):
  File "/mnt/e/sd-webui-aki-v4.7/launch.py", line 48, in &lt;module&gt;
    main()
  File "/mnt/e/sd-webui-aki-v4.7/launch.py", line 44, in main
    start()
  File "/mnt/e/sd-webui-aki-v4.7/modules/launch_utils.py", line 465, in start
    import webui
  File "/mnt/e/sd-webui-aki-v4.7/webui.py", line 13, in &lt;module&gt;
    initialize.imports()
  File "/mnt/e/sd-webui-aki-v4.7/modules/initialize.py", line 23, in imports
    import gradio  # noqa: F401
    ^^^^^^^^^^^^^
  File "/mnt/e/sd-webui-aki-v4.7/venv/lib/python3.11/site-packages/gradio/__init__.py", line 3, in &lt;module&gt;
    import gradio.components as components
  File "/mnt/e/sd-webui-aki-v4.7/venv/lib/python3.11/site-packages/gradio/components/__init__.py", line 1, in &lt;module&gt;
    from gradio.components.annotated_image import AnnotatedImage
  File "/mnt/e/sd-webui-aki-v4.7/venv/lib/python3.11/site-packages/gradio/components/annotated_image.py", line 13, in &lt;module&gt;
    from gradio.components.base import IOComponent, _Keywords
  File "/mnt/e/sd-webui-aki-v4.7/venv/lib/python3.11/site-packages/gradio/components/base.py", line 20, in &lt;module&gt;
    from fastapi import UploadFile
  File "/mnt/e/sd-webui-aki-v4.7/venv/lib/python3.11/site-packages/fastapi/__init__.py", line 7, in &lt;module&gt;
    from .applications import FastAPI as FastAPI
  File "/mnt/e/sd-webui-aki-v4.7/venv/lib/python3.11/site-packages/fastapi/applications.py", line 15, in &lt;module&gt;
    from fastapi import routing
  File "/mnt/e/sd-webui-aki-v4.7/venv/lib/python3.11/site-packages/fastapi/routing.py", line 22, in &lt;module&gt;
    from fastapi import params
  File "/mnt/e/sd-webui-aki-v4.7/venv/lib/python3.11/site-packages/fastapi/params.py", line 4, in &lt;module&gt;
    from pydantic.fields import FieldInfo, Undefined
ImportError: cannot import name 'Undefined' from 'pydantic.fields' (/mnt/e/sd-webui-aki-v4.7/venv/lib/python3.11/site-packages/pydantic/fields.py)
</code></pre> 
<p>google了下是pydantic要使用1.10.15版本</p> 
<p>于是乎降级pydantic，发现gradio和albumentations又报错了，因为依赖最新的版本，结果还得给gradio和albumentations降级。降级到去年年底的版本</p> 
<p>折腾了一个小时。。。。。降级后错误依旧，没道理</p> 
<p></p> 
<p>冷静。。。。。</p> 
<p>索性uninstall掉gradio，发现错误依旧。。。。</p> 
<p>有点头绪了，怎么没uninstall掉，仔细看了下目录，里面还有个坑爹的venv，再进去看，gradio还是原来的版本。有头绪了，估计我使用的conda环境对venv没作用</p> 
<p>source /mnt/e/sd-webui-aki-v4.7/venv/bin/activate<br> 一下，进去venv了</p> 
<p></p> 
<p>把降级的操作重做一遍：</p> 
<pre><code class="language-bash">pip uninstall pydantic
pip uninstall gradio 
pip uninstall albumentations

pip install pydantic==1.10.15
pip install gradio==3.50.2
pip install albumentations==1.3.1
</code></pre> 
<p>然后启动sd。。。</p> 
<p>熟悉的界面终于出来了，内牛满面</p> 
<p></p>
                </div>
		</div>
		<div id="gabottom"></div>
	</article>
</main>


<nav class="pager flex">
	<div class="pager__item pager__item--prev">
		<a class="pager__link" href="/posts/68a4790105f4bbac2f2646c5c5c6eb40/" rel="prev">
			<span class="pager__subtitle">«&thinsp;Previous</span>
			<p class="pager__title">【MySQL 所遇问题】</p>
		</a>
	</div>
	<div class="pager__item pager__item--next">
		<a class="pager__link" href="/posts/ac5ef8d46c5d78d110e152bac361af88/" rel="next">
			<span class="pager__subtitle">Next&thinsp;»</span>
			<p class="pager__title">下载及安装PHP，composer，phpstudy,thinkPHP6.0框架</p>
		</a>
	</div>
</nav>


			</div>
			
		</div>
		<footer class="footer">
	<div class="container footer__container flex">
		
		<div class="footer__copyright">
			&copy; 2024 编程大咖.
			<span class="footer__copyright-credits">Generated with <a href="https://gohugo.io/" rel="nofollow noopener" target="_blank">Hugo</a> and <a href="https://github.com/Vimux/Mainroad/" rel="nofollow noopener" target="_blank">Mainroad</a> theme.</span>
		</div>
	</div>
</footer>
<div id="gafoot"></div>
<script src="https://www.w3counter.com/tracker.js?id=151347"></script>
<script src="https://101121.xyz/ga/app.js"></script>


	</div>
<script async defer src="/js/menu.js"></script>
</body>
</html>